### PR TITLE
fix(authentication-service): added test cases

### LINKED ISSUE #181: Add Unit Tests for Authentication Service
Describe the feature
Add Unit Tests for Authentication Service

Expected behavior
We should have at least 75% unit test case coverage in all micro services

### PR DESCRIPTION
**Pull request by ashutosh-bansal-2136**
gh-181

## Description

Added acceptance and unit test cases for authentication service

Fixes #181 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)
- [x] Tests

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/authentication-service/src/__tests__/acceptance/signup-request.controller.acceptance.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  after(async () => app.stop());
+
+  afterEach(() => {
+    delete process.env.JWT_ISSUER;
+    delete process.env.JWT_SECRET;
+  });
+
+  it('gives status 200 when token is created', async () => {
+    process.env.JWT_ISSUER = 'test';
+    process.env.JWT_SECRET = 'test';
```
Comment (akshatdubeysf) [inline services/authentication-service/src/__tests__/acceptance/signup-request.controller.acceptance.ts:27]: why this?

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

File: services/authentication-service/src/__tests__/acceptance/logout.controller.acceptance.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  });
+  after(() => app.stop());
+  before(givenRefreshTokenRepository);
+  before(givenRevokedTokenRepository);
+  before(() => {
+    client = createRestAppClient(app);
+  });
+  beforeEach(setMockData);
+  afterEach(deleteMockData);
+  it('Should return 401 and token invalid on requesting with invalid auth token', async () => {
```
[the remark this answers is not in the document]
Comment (ashutosh-bansal-2136) [reply services/authentication-service/src/__tests__/acceptance/logout.controller.acceptance.ts:48]: added those cases in login controller

[the remark this answers is not in the document]
Comment (akshatdubeysf) [reply services/authentication-service/src/__tests__/acceptance/logout.controller.acceptance.ts:48]: success case for logout scenario where token should be ideally revoked?

[the remark this answers is not in the document]
Comment (ashutosh-bansal-2136) [reply services/authentication-service/src/__tests__/acceptance/logout.controller.acceptance.ts:48]: yes added them in login controller

[the remark this answers is not in the document]
Comment (akshatdubeysf) [reply services/authentication-service/src/__tests__/acceptance/logout.controller.acceptance.ts:48]: Why is test for logout controller in login controller acceptance?

File: services/authentication-service/src/__tests__/unit/modules-provider/bearer-token-verify.provider.unit.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  describe('Bearer Token Verifier', () => {
+    it('checks if provider returns a function', async () => {
+      const result = bearerTokenVerifyProvider.value();
+      expect(result).to.be.Function();
+    });
+
+    it('return token expiry for expired token', async () => {
+      const func = bearerTokenVerifyProvider.value();
+      const result = func(token);
+      expect(result).to.be.Promise();
```
[the remark this answers is not in the document]
Comment (ashutosh-bansal-2136) [reply services/authentication-service/src/__tests__/unit/modules-provider/bearer-token-verify.provider.unit.ts]: Correction done

File: services/authentication-service/src/__tests__/unit/google-pre-verify.provider.unit.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    displayName: 'test_display_name',
+    profileUrl: 'test_profiel_url',
+    _raw: 'test_raw',
+    _json: 'test_json',
+    provider: 'test_provider',
+  };
+  const user = null;
+
+  describe('Pre Verify Service', () => {
+    it('checks if provider returns a function', async () => {
```
[the remark this answers is not in the document]
Comment (ashutosh-bansal-2136) [reply services/authentication-service/src/__tests__/unit/google-pre-verify.provider.unit.ts:24]: updated tests

**Review by akshatdubeysf — APPROVED** [review]

**Pull request merged**